selfdrive/ui/ui.h

```diff
@@ -53,22 +60,31 @@ typedef struct Alert {
   bool equal(const Alert &a2) {
     return text1 == a2.text1 && text2 == a2.text2 && type == a2.type;
   }
-} Alert;
-
-const Alert CONTROLS_WAITING_ALERT = {"openpilot Unavailable", "Waiting for controls to start",
-                                      "controlsWaiting", cereal::ControlsState::AlertSize::MID,
-                                      AudibleAlert::NONE};
-
-const Alert CONTROLS_UNRESPONSIVE_ALERT = {"TAKE CONTROL IMMEDIATELY", "Controls Unresponsive",
-                                           "controlsUnresponsive", cereal::ControlsState::AlertSize::FULL,
-                                           AudibleAlert::CHIME_WARNING_REPEAT};
-const int CONTROLS_TIMEOUT = 5;
 
-const int bdr_s = 30;
-const int header_h = 420;
-const int footer_h = 280;
-
-const int UI_FREQ = 20;   // Hz
+  static std::optional<Alert> get(const SubMaster &sm, uint64_t started_frame) {
```

Comment:
what's the advantage of an std::optional here over returning an empty alert? I prefer avoiding std::optional because it's unclear what's going on in these lines until you read the Alert::get definition:
```
  if (auto alert = Alert::get(*(s.sm), s.scene.started_frame)) {
     if (alert->type == "controlsUnresponsive") {
     alerts->updateAlert(*alert, bgColor);
```

Reply:
reverted to std::optional.
we can't know whether we have received a controlsState message by returning an empty alert.

Reply:
why is that important? SubMaster should be used for that